dubbo-common/src/main/java/org/apache/dubbo/common/extension/support/ActivateComparator.java

```diff
@@ -44,75 +44,85 @@ public int compare(Object o1, Object o2) {
             return 0;
         }
 
-        // to support com.alibab.dubbo.common.extension.Activate
-        String[] a1Before, a2Before, a1After, a2After;
-        int a1Order, a2Order;
-        Class<?> inf = null;
-        if (o1.getClass().getInterfaces().length > 0) {
-            inf = o1.getClass().getInterfaces()[0];
+        Class<?> inf = findSpi(o1.getClass());
 
-            if (inf.getInterfaces().length > 0) {
-                inf = inf.getInterfaces()[0];
-            }
-        }
+        ActivateInfo a1 = parseActivate(o1.getClass());
+        ActivateInfo a2 = parseActivate(o2.getClass());
 
-        Activate a1 = o1.getClass().getAnnotation(Activate.class);
-        if (a1 != null) {
-            a1Before = a1.before();
-            a1After = a1.after();
-            a1Order = a1.order();
-        } else {
-            com.alibaba.dubbo.common.extension.Activate oa1 = o1.getClass().getAnnotation(com.alibaba.dubbo.common.extension.Activate.class);
-            a1Before = oa1.before();
-            a1After = oa1.after();
-            a1Order = oa1.order();
-        }
-        Activate a2 = o2.getClass().getAnnotation(Activate.class);
-        if (a2 != null) {
-            a2Before = a2.before();
-            a2After = a2.after();
-            a2Order = a2.order();
-        } else {
-            com.alibaba.dubbo.common.extension.Activate oa2 = o2.getClass().getAnnotation(com.alibaba.dubbo.common.extension.Activate.class);
-            a2Before = oa2.before();
-            a2After = oa2.after();
-            a2Order = oa2.order();
-        }
-        if ((a1Before.length > 0 || a1After.length > 0
-                || a2Before.length > 0 || a2After.length > 0)
-                && inf != null && inf.isAnnotationPresent(SPI.class)) {
+        if ((a1.before.length > 0 || a1.after.length > 0 || a2.before.length > 0 || a2.after.length > 0) && inf != null) {

```

Comment:
can be refactore by extractive to new util method e.g. 
```
if( inf !=null && (isValidActiveInfo(a1) || isValidActiveInfo(a2)) {
...
}
```
```
private boolean isValidActiveInfo(ActiveInfo activeInfo) {
   return activeInfo.before.length > 0 || activeInfo.after.length > 0
}
```

Will increase the readability. What do you say?

Reply:
done.